File: lib/src/models/fragment.dart
// lib/src/models/fragment.dart
enum FragmentType { thread, image, record, note, audio }

class DamageSpec {
  final String profile;
  final double intensity;
  final int seed;

  const DamageSpec({
    required this.profile,
    required this.intensity,
    required this.seed,
  });

  static const allowedProfiles = {
    'block-loss',
    'scanline-tear',
    'partial-decode',
    'chroma-bleed',
    'overwrite',
  };

  @override
  bool operator ==(Object other) =>
      other is DamageSpec &&
      other.profile == profile &&
      other.intensity == intensity &&
      other.seed == seed;

  @override
  int get hashCode => Object.hash(profile, intensity, seed);
}

class Fragment {
  final String id;
  final FragmentType type;
  final String label;
  final DamageSpec damage;
  final Map<String, dynamic> content;

  const Fragment({
    required this.id,
    required this.type,
    required this.label,
    required this.damage,
    required this.content,
  });
}

File: Sources/CarveCore/Models/Fragment.swift
public enum FragmentType: String, Codable, CaseIterable, Sendable {
  case thread, image, record, note, audio
}

public struct DamageSpec: Codable, Equatable, Sendable {
  public let profile: String
  public let intensity: Double
  public let seed: Int

  public static let allowedProfiles: Set<String> = [
    "block-loss", "scanline-tear", "partial-decode", "chroma-bleed", "overwrite",
  ]
}

public struct Fragment: Codable, Equatable, Sendable {
  public let id: String
  public let type: FragmentType
  public let label: String
  public let damage: DamageSpec
  /// Declarative unlock gate. Nil means visible when listed on the sector map
  /// (or always once reachability has admitted it). Evaluated via the six
  /// predicates in `Predicate.swift` — never an expression string (INV-5).
  public let hiddenUntil: [String: JSONValue]?
  public let content: [String: JSONValue]

  public init(
    id: String,
    type: FragmentType,
    label: String,
    damage: DamageSpec,
    hiddenUntil: [String: JSONValue]? = nil,
    content: [String: JSONValue]
  ) {
    self.id = id
    self.type = type
    self.label = label
    self.damage = damage
    self.hiddenUntil = hiddenUntil
    self.content = content
  }
}
